site-src/geps/gep-1762.md

```diff
@@ -0,0 +1,309 @@
+  This MUST be derived from the referenced `Spec.Address`.
+* MUST not deploy any resources into the cluster; it is expected that a user will do these actions.
+
+### Merging Gateways
+
+The spec currently specifies that gateways *can* be merged.
+If an implementation supports this, it MUST be implemented by referencing the `Gateway` to merge into.
+
+For example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: merged-gateway
+spec:
+  infrastructure:
+    attachTo:
+      kind: Gateway
+      name: parent-gateway
+  gatewayClassName: example
+  listeners:
+  - name: some-other-route
+    port: 81
+    protocol: HTTP
+```
+
+With this configuration, an implementation:
+* MUST treat the `parent-gateway` as having the concatenated list of all listeners from merged Gateways.
+  This means, for example, that a Service with port 80 and 81 may be created in this example.
+
+This introduces a new concept in Gateway merging of a "primary" Gateway.
+This makes it explicit which is the primary gateway and which are attached to it.
+
+Secondary Gateways cannot configure `addresses` or anything in `infrastructure` (aside from `attachTo`), and must have
+the same `gatewayClassName` as the primary Gateway they are attached to.
+
+Gateways can only `attachTo` a primary Gateway. That is, we cannot have a chain of references.
+
+### Customizations
+
+With any in-cluster deployment, customization requirements will arise. 
+
+Some common requirements would be:
+* `Service.spec.type`, to control whether a service is a `ClusterIP` or `LoadBalancer`.
+* IP in the Service to assign to it.
+* Arbitrary labels and annotations on generated resources.
+* Any other arbitrary fields; the list is unbounded. Some examples would be:
+  * CPU and memory requests
+  * Service `externalTrafficPolicy`
+  * Affinity rules.
+
+#### Gateway Type
+
+This is handled by [GEP-1651](https://github.com/kubernetes-sigs/gateway-api/pull/1653), so won't be described here.
+
+#### Gateway IP
+
+For setting a specific IP address in a generated `Service`, the `.Spec.Address` field can be used.
+
+For example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: my-gateway
+spec:
+  addresses:
+  - type: IPAddress
+    value: 1.1.1.1
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+```
+
+This would generate a `Service` with `clusterIP` or `loadBalancerIP`, depending on the Service type.
+
+Note: this differs from the selection of "manual deployment", which doesn't use `IPAddress` type.
+
+#### Labels and Annotations
+
+Labels and annotations SHOULD be copied from the `Gateway` to generated resources.
```

Comment:
This feels like it may lead to some undesired behaviors. Although I can see the rationale for this in some cases, I'd hate to include the suggestion in the spec unless we were very confident about it.

Reply:
Yeah, agreed. It's fine for implementations to choose to do it, but I wouldn't want it to be any stronger than a _may_.